tests/suites/test_suite_ssl.function

```diff
@@ -2155,16 +2178,22 @@ void ssl_serialize_session_save_buf_size(int ticket_len, char *crt_file,
     /* Prepare dummy session and get serialized size */
     ((void) endpoint_type);
     ((void) tls_version);
+    ((void) ticket_len);
+    ((void) crt_file);
 #if defined(MBEDTLS_SSL_PROTO_TLS1_3)
     if (tls_version == MBEDTLS_SSL_VERSION_TLS1_3) {
         TEST_ASSERT(mbedtls_test_ssl_tls13_populate_session(
                         &session, 0, endpoint_type) == 0);
-    } else
+    }
 #endif
-    {
+
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2)
+    if (tls_version == MBEDTLS_SSL_VERSION_TLS1_2) {
         TEST_ASSERT(mbedtls_test_ssl_tls12_populate_session(
                         &session, ticket_len, crt_file) == 0);
     }
+#endif
```

Comment:
If we merge these two blocks into one like this, will it be better (And we can add more such as tls1_4, etc. into one code block)?


if (tls_version == MBEDTLS_SSL_VERSION_TLS1_3) {
    #if defined(MBEDTLS_SSL_PROTO_TLS1_3)
        TEST_ASSERT(mbedtls_test_ssl_tls13_populate_session(&session, 0, endpoint_type) == 0);
    #endif
}
else if (tls_version == MBEDTLS_SSL_VERSION_TLS1_2) {
    #if defined(MBEDTLS_SSL_PROTO_TLS1_2)
        TEST_ASSERT(mbedtls_test_ssl_tls12_populate_session(&session, ticket_len, crt_file) == 0);
    #endif
}

Reply:
Change to `switch .. case `